dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/transfer/DataTransferServicesImpl.java

```diff
@@ -108,22 +114,37 @@ public DataTransferServicesImpl(
     }
 
     @Override
-    public String doExport() throws java.io.IOException {
-        String zipLocation = new StringBuilder()
-            .append(System.getProperty("java.io.tmpdir"))
-            .append(File.separator)
-            .append(FILE_PATH)
-            .append(File.separator)
-            .append(EXPORT_FILE_NAME)
-            .toString();
+    public String doExport(DataTransferExportModel exportModel) throws java.io.IOException {
+        String zipLocation = new StringBuilder().append(TMP_DIR)
+                                                .append(File.separator)
+                                                .append(FILE_PATH)
+                                                .append(File.separator)
+                                                .append(EXPORT_FILE_NAME)
+                                                .toString();
+
+        Predicate<Path> readmeFilter = p -> p.toString().toLowerCase().endsWith("readme.md");
+        Predicate<Path> datasetsFilter = def -> true;
+        Predicate<Path> pagesFilter = page -> true;
+        boolean exportNavigation = true;
+
+        if (!exportModel.isExportAll()) {
+            datasetsFilter = filterDatasets(exportModel.getDatasetDefinitions());
+            pagesFilter = filterPages(exportModel.getPages());
+            exportNavigation = exportModel.isExportNavigation();
```

Comment:
I would let it in case we want to allow users to select this option in future. For example, I did create [1] so users could export a single page along its dependencies so they can it on runtime. In that case navigation would not be needed.

[1] https://issues.redhat.com/browse/AF-2459